docs/testplan/HA-SmartSwitch-test-plan.md

```diff
@@ -0,0 +1,171 @@
+
+| Date       | Author        | Description                      |
+|------------|---------------|----------------------------------|
+| 2024-02-06 | Jing Zhang   | Initial draft                    |
+
+## Table of Contents
+
+- [Smart Switch HA – Test Plan](#smart-switch-ha--test-plan)
+  - [Revision History](#revision-history)
+  - [Scope](#scope)
+  - [Topology](#topology)
+  - [Test Methodology](#test-methodology)
+  - [Test Plan](#test-plan)
+    - [Downstream (T2->T0) Traffic Verification](#downstream-t2t0-traffic-verification)
+      - [Module 1 	Normal OP and Planned Events](#module-1--normal-op-and-planned-events)
+      - [Module 2 	BFD state pinned](#module-2--bfd-state-pinned)
+      - [Module 3 	HA state pinned by upstream service](#module-3--ha-state-pinned-by-upstream-service)
+      - [Module 4 	Link Failures](#module-4--link-failures)
+      - [Module 5 	Critical Process Crash](#module-5--critical-process-crash)
+      - [Module 6 	Power down and hardware failure](#module-6--power-down-and-hardware-failure)
+      - [Module 7 	Operations](#module-7--operations)
+    - [Upstream (T0->T2) Traffic Verification](#upstream-t0t2-traffic-verification)
+      - [Module 1 Normal OP](#module-1-normal-op)
+  - [Test Utilities](#test-utilities)
+
+## Scope 
+This document proposes solutions for Smart Switch High-Availability test plans. The document will cover smart switch test scenarios.
+
+The goal of this test plan is to verify HA state machine behavior in normal operation scenarios and network failure scenarios with [t1-smartswitch-ha](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/vars/topo_t1-smartswitch-ha.yml) topology. Both control plane and data plane will need to be verified in the test cases. 
+
+## Topology
+
+Tests will run on a sonic-mgmt testbed. The key components will be:
+1. Test server
+1. Fanout switches
+    * root fanout switch
+    * leaf fanout switch
+1. SONiC DUTs -- the two smartswitches.
+
+Key aspects of the physical connection:
+1. Every DUT port is connected to the leaf fanout switch.
+1. Every leaf fanout switch has unique VLAN tag for every DUT port.
+1. Root fanout switch connects leaf fanout switches and test servers using 802.1Q trunks. 
+
+Any test server can access any DUT port by sending a packet with the port VLAN tag. In test servers, a set of dockers can be created for running cEOS to simulate neighbors of the SONiC DUT. A PTF container (based on the PTF test framework) for injecting/sniffing packets will be created. The PTF docker, the cEOS dockers and VLAN interfaces in test server can be interconnected by open vSwitch bridges. 
+
+![physical-connection](./Img/topo_smartswitch_ha.png)
+
+Please refer to [sonic-mgmt testbed overview](https://github.com/sonic-net/sonic-mgmt/blob/master/docs/testbed/README.testbed.Overview.md) for more about testbed setup and diagrams. 
+
+
+## Test Methodology
+
+1. Open flow rules are configured on OVS Bridges to control the traffic flow:
```

Comment:
this looks still target to VM testbed only. do we need to get this updated, so we can get the others captured too?

Reply:
This is still need for physical testbeds, as we are using cEOS neighbors and ptf docker